src/components/table/table-row/table-row.style.js

```diff
@@ -1,166 +1,148 @@
 
 /**
  * Current version of react-dnd used in DragAndDrop is incompatible
  * with styled-components, this can be uodated once the issue is fixed
  */
-const StyledTableRow = css`
-  .carbon-table-row {
-    ${applyRowStyling}
-  }
+const StyledTableRow = styled.tr`
+  ${applyRowStyling}
 `;
 
-function applyRowStyling(props) {
-  const { theme } = props;
+function applyRowStyling({ isPassive, isSelected, theme }) {
   return css`
-    .custom-drag-layer {
-      & {
-        background-color: #002E41;
-        cursor: grabbing;
-        cursor: -moz-grabbing;
-        cursor: -webkit-grabbing;
-        display: block;
-      }
-
-      ${StyledTableCell} {
-        background-color: #002E41;
-      }
-
-      .configurable-item-row__content-wrapper {
-        visibility: visible;
-      }
-    }
-
-    &:first-child ${StyledTableHeader} {
-      &:first-child {
-        border-radius: 0px 0 0 0;
-      }
-
-      &:last-child {
-        border-radius: 0 0px 0 0;
-      }
-    }
-
-    ${isClassic(theme) ? applyClassicRowStyling() : applyModernRowStyling(theme)}
+    ${isClassic(theme) ? applyClassicRowStyling(isPassive, isSelected) : applyModernRowStyling(isPassive, theme)}
     ${selectableRowStyling}
-    ${selectedRowStyling}
     ${highlightRowStyling}
+    ${selectedRowStyling}
     ${dragRowStyling}
   `;
 }
 
-function selectableRowStyling({ theme }) {
+function selectableRowStyling({ isSelectable, theme }) {
   return css`
-    ${StyledTableCell}:first-child,
-    ${StyledTableHeader}:first-child {
-      &:not(.carbon-table-cell--select) {
-        ${isClassic(theme) ? 'padding-left: 15px' : ''};
+    ${isSelectable && css`
+      ${StyledTableCell}:first-child,
+      ${StyledTableHeader}:first-child {
+        &:not(['data-component=selectable-cell']) {
+          ${isClassic(theme) ? 'padding-left: 15px' : ''};
+        }
       }
-    }
 
-    .carbon-table-cell--select:first-child,
-    carbon-table-cell--select:first-child {
-        text-align: center;
+      ${StyledTableCell}:first-child {
         width: 18px;
-      
-        .carbon-checkbox {
-          height: 15px;
-          padding-top: 0;
-        }
       }
-    }
+
+      ${CheckboxStyle} {
+        height: 15px;
+        padding-top: 0;
+      }
+    `}
   `;
 }
 
-function selectedRowStyling({ theme }) {
+function selectedRowStyling({ isSelected, theme }) {
   return css`
-    .carbon-table-row--selected,
-    .carbon-table-row--selected:nth-child(odd),
-    .carbon-table-row--selected:hover {
-      ${StyledTableCell} {
-        ${isClassic(theme) ? applyClassicSelectedStyling(theme) : applyModernSelectedStyling(theme)}
-        position: relative;
-    
-        &:before {
-          content: "";
-          height: 1px;
-          left: 0;
-          position: absolute;
-          top: -1px;
-          width: 100%;
+    ${isSelected && css`
+      &&&&,
+      &&&&:nth-child(odd),
+      &&&&:hover {
```

Comment:
Is there any other option than do `&&&&`?

Reply:
I don't think so :disappointed:

@stevebutler2210 implemented this in 23a1e4a, and I believe he tried every possible alternative solution he could think of.

I think the problem is that the table row is styled with a high specificity — `StyledTable & StyledTableRow &:nth-child(2n)` — in `table.style.js` and `table-modern.style.js`, so we need to use [an even higher specificity here with `&&&&`](https://www.styled-components.com/docs/faqs#how-can-i-override-styles-with-higher-specificity) to make it work.